packages/frontend-api/src/Model/Resolver/Products/ProductsQuery.php

```diff
@@ -226,4 +205,106 @@ function ($offset, $limit) use ($argument, $category, $productFilterData, $order
             $readyCategorySeoMix,
         );
     }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @param \Shopsys\FrameworkBundle\Model\Product\Flag\Flag $flag
+     * @return \GraphQL\Executor\Promise\Promise
+     */
+    public function productsByFlagQuery(Argument $argument, Flag $flag): Promise
+    {
+        PageSizeValidator::checkMaxPageSize($argument);
+
+        $this->setDefaultFirstOffsetIfNecessary($argument);
+
+        $productFilterData = $this->productFilterFacade->getValidatedProductFilterDataForFlag(
+            $argument,
+            $flag,
+        );
+
+        $productFilterData->flags[] = $flag;
+        $batchLoadDataId = Uuid::uuid4()->toString();
+
+        return $this->productConnectionFactory->createConnectionPromiseForFlag(
+            $flag,
+            function ($offset, $limit) use ($argument, $productFilterData, $flag, $batchLoadDataId) {
+                return $this->productsByEntitiesBatchLoader->load(
+                    new ProductBatchLoadByEntityData(
+                        $batchLoadDataId,
+                        $flag->getId(),
+                        Flag::class,
+                        $limit,
+                        $offset,
+                        $this->productOrderingModeProvider->getOrderingModeFromArgument($argument),
+                        $productFilterData,
+                    ),
+                );
+            },
+            $argument,
+            $productFilterData,
+            $this->productOrderingModeProvider->getOrderingModeFromArgument($argument),
+            $this->productOrderingModeProvider->getDefaultOrderingMode($argument),
+            $batchLoadDataId,
+        );
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @param \GraphQL\Type\Definition\ResolveInfo $info
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Connection\ProductConnection|\GraphQL\Executor\Promise\Promise
+     */
+    public function productsWithOverlyingEntityQuery(Argument $argument, ResolveInfo $info): ProductConnection|Promise
```

Comment:
there is still `project-base/app/config/graphql/types/Query/Product/ProductQuery.types.yaml` where this query is defined (and overrides the `products` query from `QueryDecorator`, could you move that as well, please?

Reply:
moved in c2a8cb8461